library/authentication/password_change.php

```diff
@@ -121,9 +121,9 @@ function update_password($activeUser, $targetUser, &$currentPwd, &$newPwd, &$err
 
     // End active user check
 
-    
-    //Test password validity
-    if (strlen($newPwd) == 0 && empty($GLOBALS['gbl_ldap_enabled'])) {
+
+    //Test password validity (note that even for ldap a random password is created above)
```

Comment:
Not sure I understand this comment. I guess it's OK to allow changing the OpenEMR password with LDAP in use since you might turn off LDAP later, or disable it for the user.

Reply:
Just pointing out that $newPwd is set above when using LDAP (it will not be empty) to a random password (I changed the random mechanism above to by cryptographically secure), so no need to explicitly bypass this via the ldap global. From my understanding in LDAP, this will only be used when creating a new user (it will place a random password in openemr that will never be used; it's a good idea though in case LDAP gets turned off).